bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/operation/DetectorsTransmissionVisitor.java

```diff
@@ -77,26 +81,38 @@ private void initialize(final double[][] data) {
         int finalPixelValue = (int) data[0][data[0].length - 1];
         double[] pixelData = new double[finalPixelValue];
         pixelData[0] = 1.0; // XXX previously defaulted to 0
-        int j = 0;
+        int j = 0;                                    // ccdpix data index
         detectorCcdIndexes = new ArrayList<>(6);
         detectorCcdIndexes.add(0);
         int prevY = 1;
-        for (int i = 1; i < finalPixelValue; i++) {
-            int x = (int) data[0][j];
-            int y = (int) data[1][j];
-            if (prevY != y) {
+        for (int i = 1; i < finalPixelValue; i++) {   // pixel index
+            int x = (int) data[0][j];                 // ccdpix wavelength
+            int y = (int) data[1][j];                 // ccdpix transmission
+            if (prevY != y) {                         // if the transmission changes then this is a CCD boundary
                 detectorCcdIndexes.add(x - prevY - 1);
                 prevY = y;
             }
-            if (x == i) {
+            if (x == i) {              // if this pixel is in ccdpix use its transmission value
                 pixelData[i] = y;
                 j++;
-            } else {
-                // Apply the last transmission pixel value and go on without incrementing
+            } else {                   // else use the last transmission value and go on without incrementing
                 pixelData[i] = data[1][j - 1];
             }
         }
         detectorCcdIndexes.add(finalPixelValue - 1);
+        Log.fine("detectorCcdIndexes = " + detectorCcdIndexes);
+
+        // If the user has selected ROI = CCD2 then set the transmission of pixels in CCD1 and CCD3 to zero:
+        if (ROI == GmosCommonType.BuiltinROI.CCD2) {
```

Comment:
There are three choices for ROI but it seems only CCD2 has a special treatment, Is that right?

Reply:
Thanks for pointing this out.  Both CCD2 and CENTRAL_STAMP affect the output, and I have now included the effect of both.